### ISSUE TITLE
CODEP在conala数据集上的BLEU表现

### ISSUE THREAD

**Issue by baday19** [YihongDong/CODEP#1]
您好，我使用您的代码在conala数据集上进行了测试，但结果不太符合预期，不知道是否是参数设置有误，所以有一些问题想要咨询您。以下是我的运行过程和参数设置：

1. 将conala数据集放置到了对应的目录后运行了gendata以获得处理后的数据集及词典

2. 运行scripts/conala目录下的脚本

   **transformer+pda(直接运行train_transformer.sh):**

   参数：
`Namespace(NAME_TOKEN_NUM=15, action_embed_size=256, alpha=1.0, answer_prune=True, att_vec_size=256, attn_heads=4, batch_size=128, beam_size=15, clip_grad=5.0, code_token_label_smoothing=0.0, column_att='affine', cuda=True, decay_lr_every_epoch=False, decode_max_time_step=50, decoder_layers=2, decoder_word_dropout=0.3, dev_decode_file=None, dev_file='data/conala/dev.bin', dropout=0.1, embed_size=256, encoder_layers=2, eval_top_pred_only=False, evaluator='default_evaluator', example_preprocessor=None, features=None, feed_nonterminal=True, glorot_init=True, glove_embed_path=None, grammar_version='3.7', hidden_size=512, load_decode_results=None, load_model=None, load_paraphrase_model=None, load_reconstruction_model=None, load_reranker=None, log_every=50, lr=0.0001, lr_decay=0.5, lr_decay_after_epoch=600, lstm='lstm', max_epoch=600, max_num_trial=5, metric='accuracy', mode='train', negative_sample_type='best', no_copy=False, no_input_feed=False, no_query_vec_to_action_map=False, num_workers=1, optimizer='Adam', parser='transformer_parser', patience=5, predict_with_nonterminal=True, pretrain=None, ptrnet_hidden_dim=32, query_vec_to_action_diff_map=False, readout='identity', reset_optimizer=False, sample_size=5, save_all_models=False, save_decode_to='decodes/conala/conala.transformer.parse.a1.d0.1.bs128.lr0.0001.hs512.es256.aes256.NAME_TOKEN_NUM15.me600.un.pwn.fd.test.decode', save_to='saved_models/conala/conala.transformer.parse.a1.d0.1.bs128.lr0.0001.hs512.es256.aes256.NAME_TOKEN_NUM15.me600.un.pwn.fd', seed=0, sql_db_file=None, src_token_label_smoothing=0.0, state_embed_size=128, sup_attention=False, test_decode_file=None, test_file='data/conala/test.bin', tie_embed=False, train_decode_file=None, train_file='data/conala/train.bin', uniform_init=None, unlabeled_file=None, unsup_loss_weight=1.0, use_nonterminal=True, vaildate_begin_epoch=200, valid_every_epoch=10, valid_metric='acc', verbose=False, vocab='data/conala/vocab.src_freq2.code_freq2.bin', warm_up_change_every=2, warm_up_step=10, word_dropout=0.0)`

   结果(yours: 27.87±0.9)：

   ```json
   {'corpus_bleu': 0.25134014969478413, 'oracle_corpus_bleu': 0.3617340385743742, 'avg_sent_bleu': 0.20721409401394883, 'oracle_avg_sent_bleu': 0.3569642105012174, 'exact_match': 0.026, 'oracle_exact_match': 0.066}
   ```

   **lstm+pda(直接运行train.sh):**

   参数：
  `Namespace(NAME_TOKEN_NUM=15, action_embed_size=256, alpha=1.0, answer_prune=True, att_vec_size=256, attn_heads=4, batch_size=32, beam_size=15, clip_grad=5.0, code_token_label_smoothing=0.0, column_att='affine', cuda=True, decay_lr_every_epoch=False, decode_max_time_step=50, decoder_layers=2, decoder_word_dropout=0.3, dev_decode_file=None, dev_file='data/conala/dev.bin', dropout=0.5, embed_size=256, encoder_layers=2, eval_top_pred_only=False, evaluator='default_evaluator', example_preprocessor=None, features=None, feed_nonterminal=True, glorot_init=True, glove_embed_path=None, grammar_version='3.7', hidden_size=512, load_decode_results=None, load_model=None, load_paraphrase_model=None, load_reconstruction_model=None, load_reranker=None, log_every=50, lr=0.001, lr_decay=0.5, lr_decay_after_epoch=60, lstm='lstm', max_epoch=200, max_num_trial=5, metric='accuracy', mode='train', negative_sample_type='best', no_copy=False, no_input_feed=False, no_query_vec_to_action_map=False, num_workers=1, optimizer='Adam', parser='lstm_parser', patience=5, predict_with_nonterminal=True, pretrain=None, ptrnet_hidden_dim=32, query_vec_to_action_diff_map=False, readout='identity', reset_optimizer=False, sample_size=5, save_all_models=False, save_decode_to='decodes/conala/conala.parse.a1.d0.5.bs32.lr0.001.hs512.es256.aes256.NAME_TOKEN_NUM15.me200.un.pwn.fd.test.decode', save_to='saved_models/conala/conala.parse.a1.d0.5.bs32.lr0.001.hs512.es256.aes256.NAME_TOKEN_NUM15.me200.un.pwn.fd', seed=0, sql_db_file=None, src_token_label_smoothing=0.0, state_embed_size=128, sup_attention=False, test_decode_file=None, test_file='data/conala/test.bin', tie_embed=False, train_decode_file=None, train_file='data/conala/train.bin', uniform_init=None, unlabeled_file=None, unsup_loss_weight=1.0, use_nonterminal=True, vaildate_begin_epoch=20, valid_every_epoch=1, valid_metric='acc', verbose=False, vocab='data/conala/vocab.src_freq2.code_freq2.bin', warm_up_change_every=1, warm_up_step=0, word_dropout=0.0)`

   结果(yours: 28.22±0.8)：

   ```json
   {'corpus_bleu': 0.2516496774445468, 'oracle_corpus_bleu': 0.355866360767733, 'avg_sent_bleu': 0.2057969390941255, 'oracle_avg_sent_bleu': 0.32706856364565634, 'exact_match': 0.024, 'oracle_exact_match': 0.038}
   ```

主要的点在于BLEU上，似乎与您的结果存在一定的差距。您能帮我看看么，十分感谢！

**Comment by YihongDong** [comment]
抱歉，是否有可能是环境的原因，我当时使用的是Python 3.7.11以及torch 1.10.1

以下是我训练时的wandb，可以作为参考：
https://wandb.ai/dongyh/parse-conala/reports/LSTM-PDA-CONALA---Vmlldzo0OTg2OTEw
